test/unit/graphql/metrics/integration_test.rb

```diff
@@ -219,6 +219,18 @@ def self.parse_error(err, _context)
         assert_equal_with_diff_on_failure(kitchen_sink_expected_arguments, kitchen_sink_query_metrics[:arguments])
       end
 
+      test "safely skips logging arguments metrics for fields, when the argument value look up fails (possibly because it failed input coercion)" do
+        GraphQL::Query.any_instance.stubs(:arguments_for).raises(GraphQL::ExecutionError.new(nil))
+
+        query = GraphQL::Query.new(
+          SchemaWithFullMetrics,
+          kitchen_sink_query_document,
+          variables: { 'postId': '1', 'titleUpcase': true },
+          operation_name: 'PostDetails',
+        )
+        result = query.result.to_h
```

Comment:
More to assert below, like that we still logged usage of fields, but simply skipped logging args metrics.

Reply:
At the moment, though: this fails with 

```
  1) Error:
GraphQL::Metrics::IntegrationTest#test_safely_skips_logging_arguments_metrics_for_fields,_when_the_argument_value_look_up_fails_(possibly_because_it_failed_input_coercion):
TypeError: nil can't be coerced into Float
    /Users/chris/src/github.com/Shopify/graphql-metrics/lib/graphql/metrics/instrumentation.rb:31:in `-'
    /Users/chris/src/github.com/Shopify/graphql-metrics/lib/graphql/metrics/instrumentation.rb:31:in `after_query'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:82:in `public_send'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:82:in `block in call_after_hooks'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:80:in `each'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:80:in `call_after_hooks'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:74:in `ensure in call_hooks'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:75:in `call_hooks'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:44:in `each_query_call_hooks'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:27:in `block in apply_instrumenters'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:72:in `call_hooks'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/instrumentation.rb:26:in `apply_instrumenters'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/multiplex.rb:175:in `instrument_and_analyze'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/multiplex.rb:61:in `block in run_queries'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/tracing.rb:82:in `call_tracers'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/tracing.rb:84:in `block in call_tracers'
    /Users/chris/src/github.com/Shopify/graphql-metrics/lib/graphql/metrics/tracer.rb:49:in `trace'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/tracing.rb:84:in `call_tracers'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/tracing.rb:68:in `trace'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/execution/multiplex.rb:59:in `run_queries'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/query.rb:195:in `block in result'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/query.rb:439:in `with_prepared_ast'
    /Users/chris/.gem/ruby/2.6.5/gems/graphql-1.10.8/lib/graphql/query.rb:194:in `result'
    /Users/chris/src/github.com/Shopify/graphql-metrics/test/unit/graphql/metrics/integration_test.rb:231:in `block in <class:IntegrationTest>'
```

Reply:
That said, you said you tested locally with this patch, and it worked 🤔

Reply:
Hm. Can't seem to reproduce the error. Maybe I just hadn't saved something.

Reply:
Re-ran with the seed that "failed", and it's fine now.

```ruby -Itest test/unit/graphql/metrics/integration_test.rb --seed 15085```